### PR TITLE
[4.0] [Cassiopea] Implementing Password length/meter

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
As title says.

### Testing Instructions
Allow user registration in Users Options
Register in frontend Cassiopea

Uses npm after patch

### Actual result BEFORE applying this Pull Request
Screen Shot 2020-08-20 at 07 45 32

### Expected result AFTER applying this Pull Request

This will work better (length) if https://github.com/joomla/joomla-cms/pull/30424 is already merged in the testing branch.

Screen Shot 2020-08-21 at 11 53 40

There may be some cs. tbs after drone first pass.

### CODE REVIEW

**Comment by Magnytu2** [comment]
the graph is very small on Safari as on Chrome
Créer_un_compte-2

**Comment by infograf768** [comment]
This PR needs npm to add the class:
```
meter {
  width: 100%;
}
```
can't be tested with patch tester.
Also to get good results for the color bar, the test site must already contain https://github.com/joomla/joomla-cms/pull/30424

**Comment by Magnytu2** [comment]
I tried on my site with the PR 30424, but it is still small.

**Comment by infograf768** [comment]
both prs need npm to compile the scss.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 9697d5bb20a65d9e6a1a1bcedcd57478c5a7320e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 9697d5bb20a65d9e6a1a1bcedcd57478c5a7320e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

File: language/en-GB/com_users.ini [hunk 1]
```diff
@@ -48,6 +48,7 @@ COM_USERS_MAIL_FAILED="Failed sending email."
 COM_USERS_MAIL_SEND_FAILURE_BODY="An error was encountered when sending the user registration email. The error is: %s The user who tried to register is: %s"
 COM_USERS_MAIL_SEND_FAILURE_SUBJECT="Error sending email"
 COM_USERS_OR="or"
+COM_USERS_PASSWORD_LENGTH="Enter at least 12 characters."
```
Comment (zero-24) [inline language/en-GB/com_users.ini]: That value is not hardcoded so i'm not sure whether we should hardcode it into a language string.

**Review by zero-24 — CHANGES_REQUESTED** [review]

File: language/en-GB/com_users.ini (hunk 1 above)
Comment (infograf768) [reply language/en-GB/com_users.ini]: Is'nt it hardcoded
`$this->minLength    = 12;`
`$this->minLength    = (int) ComponentHelper::getParams('com_users')->get('minimum_length', 12);`

or is it only for installation?

Comment (infograf768) [reply language/en-GB/com_users.ini]: We could fetch the value from the Options here and add a variable in the string.
True minimum is 8 indeed.

What do you think?

Comment (zero-24) [reply language/en-GB/com_users.ini]: As mention in the maintainers chat already.

There is just the default to be 12 chars an no complexity rules.

Administrators can choose to enforce complxity rules including lower or higher limits in characters as well as minimum integer etc.

The core by default just have that 12 char rule that is also used in the installer as no other config is aviable at that point.

But when you want to do such string we might have to mention all configured complexity rules and not just the char count.

Comment (infograf768) [reply language/en-GB/com_users.ini]: OK. Let's take off the desc for now to get this in and then look later at a way to display by a specific icon the rules defined by the Options.

**Comment by infograf768** [comment]
Thanks @zero-24 
Took off description. We have to implement another PR to display all the minimum rules defined in the Options.

We will not just get
Screen Shot 2020-08-22 at 11 18 07

Folks, please test again

**Comment by infograf768** [comment]
Back to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

**Comment by richard67** [comment]
@zero-24 Could you update your review so that the "Changes requested" disappears? I think it needs to approve the changes of this for that, marking conversation as revolved will not be enough, as far as I remember.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on cfced3641349636b9893e96ae71fbfa98c117f23<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on cfced3641349636b9893e96ae71fbfa98c117f23<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30437">issues.joomla.org/tracker/joomla-cms/30437</a>.</sub>

**Pull request merged**